tasks/actorstate/market/deal_proposal.go

```diff
@@ -34,74 +35,71 @@ func (DealProposalExtractor) Extract(ctx context.Context, a actorstate.ActorInfo
+		}
+
 		if err := currDealProposals.ForEach(func(id abi.DealID, dp market.DealProposal) error {
-			var label string
-			if dp.Label.IsString() {
-				var err error
-				label, err = dp.Label.ToString()
-				if err != nil {
-					return fmt.Errorf("creating deal proposal label string: %w", err)
-				}
-			} else {
-				label = ""
-			}
-			out = append(out, &marketmodel.MarketDealProposal{
-				Height:               int64(ec.CurrTs.Height()),
-				DealID:               uint64(id),
-				StateRoot:            ec.CurrTs.ParentState().String(),
-				PaddedPieceSize:      uint64(dp.PieceSize),
-				UnpaddedPieceSize:    uint64(dp.PieceSize.Unpadded()),
-				StartEpoch:           int64(dp.StartEpoch),
-				EndEpoch:             int64(dp.EndEpoch),
-				ClientID:             dp.Client.String(),
-				ProviderID:           dp.Provider.String(),
-				ClientCollateral:     dp.ClientCollateral.String(),
-				ProviderCollateral:   dp.ProviderCollateral.String(),
-				StoragePricePerEpoch: dp.StoragePricePerEpoch.String(),
-				PieceCID:             dp.PieceCID.String(),
-				IsVerified:           dp.VerifiedDeal,
-				Label:                SanitizeLabel(label),
+			dealProposals = append(dealProposals, market.ProposalIDState{
+				ID:       id,
+				Proposal: dp,
 			})
 			return nil
 		}); err != nil {
-			return nil, fmt.Errorf("walking current deal states: %w", err)
+			return nil, err
+		}
+	} else {
+		// else diff the actor against previous state and collect any additions that occurred.
+		changed, err := ec.CurrState.ProposalsChanged(ec.PrevState)
+		if err != nil {
+			return nil, fmt.Errorf("checking for deal proposal changes: %w", err)
+		}
+		if !changed {
+			return nil, nil
 		}
-		return out, nil
 
-	}
+		changes, err := market.DiffDealProposals(ctx, ec.Store, ec.PrevState, ec.CurrState)
+		if err != nil {
+			return nil, fmt.Errorf("diffing deal proposals: %w", err)
+		}
 
-	changed, err := ec.CurrState.ProposalsChanged(ec.PrevState)
-	if err != nil {
-		return nil, fmt.Errorf("checking for deal proposal changes: %w", err)
+		for _, change := range changes.Added {
+			dealProposals = append(dealProposals, market.ProposalIDState{
+				ID:       change.ID,
+				Proposal: change.Proposal,
+			})
+		}
 	}
 
-	if !changed {
-		return nil, nil
-	}
+	out := make(marketmodel.MarketDealProposals, len(dealProposals))
+	for idx, add := range dealProposals {
+		var isString bool
+		var base64Label string
+		if add.Proposal.Label.IsString() {
+			labelString, err := add.Proposal.Label.ToString()
+			if err != nil {
+				return nil, fmt.Errorf("deal proposal (ID: %d) label is not a string despite claiming it is (developer error?)", add.ID)
+			}
 
-	changes, err := market.DiffDealProposals(ctx, ec.Store, ec.PrevState, ec.CurrState)
-	if err != nil {
-		return nil, fmt.Errorf("diffing deal states: %w", err)
-	}
+			isString = true
+			base64Label = base64.StdEncoding.EncodeToString([]byte(labelString))
```

Comment:
I think yes because the string may have utf stuff in it and make very ugly csvs (if not broken ones).

Reply:
Is the ask to sanitize the label (using the currently removed `SanitizeLabel` method) before encoding it as base64?

Reply:
Moving my comment here from Slack for posterity:

My expectation is that if I decode a string from base64, it should already be cleaned of invalid characters, otherwise this falls into the realm of "surprising and unexpected" behavior.